packages/compiler/src/process/parser/yaml/index.ts

```diff
@@ -46,9 +48,12 @@ export default class YAML implements Parser {
     const steps: (Trigger | Filter | Task)[] = []
     for (const step of process.steps) {
       if (step.type !== 'filter') {
-        const newStep = this.preprocesses
-          .filter((sub) => sub.match(step))
-          .reduce((s, sub) => sub.replace(s), step)
+        let newStep = step
+        for (const sub of this.preprocesses) {
+          if (await sub.match(step)) {
+            newStep = await sub.replace(step)
```

Comment:
I think this should be

```suggestion
            newStep = await sub.replace(newStep)
```

If not we only replace it with the latest step that is processed and not combine all the preprocessing like it was with the `reduce`.